api/v1alpha1/imageclusterinstall_webhook.go

```diff
@@ -81,20 +69,43 @@ func (r *ImageClusterInstall) ValidateUpdate(old runtime.Object) (admission.Warn
 		return nil, fmt.Errorf("old object is not an ImageClusterInstall")
 	}
 
-	if oldClusterInstall.Spec.BareMetalHostRef == nil && r.Spec.BareMetalHostRef == nil {
-		return nil, nil
-	}
-
 	// Allow update if it's not the spec
 	if !isSpecUpdate(oldClusterInstall, r) {
 		return nil, nil
 	}
-	if BMHRefsMatch(oldClusterInstall.Spec.BareMetalHostRef, r.Spec.BareMetalHostRef) {
-		return nil, fmt.Errorf("Cannot update ImageClusterInstall when BareMetalHostRef is set, unset BareMetalHostRef before making changes")
+	// block update if the installation started
+	if installationStarted(oldClusterInstall) {
+		return nil, fmt.Errorf("cannot update ImageClusterInstall when the configImage is ready")
 	}
 	return nil, nil
 }
 
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *ImageClusterInstall) ValidateDelete() (admission.Warnings, error) {
+	return nil, nil
+}
+
+func (r *ImageClusterInstall) validate() error {
+	if err := isValidSSHPublicKey(r.Spec.SSHKey); err != nil {
```

Comment:
Maybe add validation that we have referenced bmh?

Reply:
What for?

Reply:
as we don't support ici without bmh anymore? but it's ok if we allow to set it after